src/common/components/header/index.js

```diff
@@ -7,7 +7,7 @@ import styles from './header.css';
 
 const wordmark = 'https://assets.ubuntu.com/v1/d45097a4-snapcraft.io-logotype.svg';
 
-class Header extends Component {
+export class Header extends Component {
```

Comment:
@earnubs @katiefenn @nottrobin 

It seems to me that we all agree on the principle of having a clean code and good separation of concerns. Where we differ is the interpretation of how to best implement this in React/Redux world. And unfortunately (or maybe fortunately) React/Redux leave a lot of space for interpretation, because they are not very opinionated and leave to developer how you use them.

I totally agree that we want simple presentational components that only render the data and container components that connect them to data and give more functionality. And that's exactly what React/Redux allows us.

Looking at UserAvatar example:

```js
class UserAvatar extends Component {

  render() {
    const { auth, user } = this.props;

    if (!auth.authenticated) return null;

    return (
      <div className={ styles.container }>
          // render component here...
      </div>
    );
  }
}

UserAvatar.propTypes = {
  auth: PropTypes.object,
  user: PropTypes.object
};
```

This is a presentational user avatar component. All it does is rendering some HTML based on user/auth data it gets in props.

```
function mapStateToProps(state) {
  const {
    auth,
    user
  } = state;

  return {
    auth,
    user
  };
}

export default connect(mapStateToProps)(UserAvatar);
```

This is a container component that connects data from Redux store and passes it to UserAvatar presentational component to render it. That's the "dependency injection".
No other component or container in the app, even if it wants to render user avatar needs to know what kind of data it needs.

Clearly separated concerns in 2 separate components. They just happen to be in one file.

By default we export container component because that's what we use in the app. Presentational component is kept 'hidden' in the module and not visible to outside world.

And now we are back to testing. We didn't do much testing of React/Redux components before because we didn't have time to and when we did it was just for pure presentational components.

Should we just test the container component? Container component is a wrapper, its only functionality is passing the data from the store to presentational component. So I believe we should be able to test presentational component as well, to check if it renders correctly based on data it gets in props.

And yes, separately we can also test container component, but we don't need to check how it renders (assuming we already tested presentational component). We only need to check if it maps the state from the store to the props as presentational component expects. Because that's the function of container component.

But to be able to test them separately, we need to export them both.
Maybe it was mistake we started to put both presentational and container components in one file - but this seemed to be a convention we borrowed from the examples we found.
Should we put them in separate files in same module? That module would still need to export both to make them visible for tests.
Should we separate them into totally separate component modules? In theory it would make sense, but from practical point of view it will make our components structure totally bloated with doubled components (and we would need some good naming convention for those).